grpc.go

```diff
@@ -56,40 +58,85 @@ type onlineResource struct {
 
 var emptyResp = &gpb.Empty{}
 
-func onlineCPUMem() error {
-	resourceList := []onlineResource{
-		{
-			sysfsOnlinePath: sysfsCPUOnlinePath,
-			regexpPattern:   cpuRegexpPattern,
-		},
-		{
-			sysfsOnlinePath: sysfsMemOnlinePath,
-			regexpPattern:   memRegexpPattern,
-		},
+var onlineCPUMemMux sync.Mutex
+
+const onlineCPUMemWaitTime = 500 * time.Millisecond
```

Comment:
This looks like a lot ! We cannot afford to wait for so much time between two retries. Something like 50ms seems more appropriate, and maybe increase the number of retries if you think that in some cases it might take some time.